### PR TITLE
New Game Pieces

### PR DESCRIPTION
**Pull request by RaphaelAlvez**
Adds assets to game pieces and remove color options on add widget overlay

### This PR has one breaking change. The default color has be changed to "#ff0000" to support the new svgReplace default

- [ ]  add a color picker to the add widget overlay
- [ ] adjust positions
- [ ] add dices
- [ ] add poker chips
- [ ] add option for text on editor UI (mostly for poker chips)
- [ ] change editor UI to display a color picker whenever the image is an SVG (@robartsd and @ArnoldSmith86 I need help with this)
- [ ] remove old game pieces?
- [ ] add file updater

This PR requires documentation for svgReplace that I can refers to in the editor UI

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
Changing the default of `svgReplaces` won't be pretty BTW. For every basic widget in existence we need to add `svgReplaces` with the old defaults or look at the `image` first and only do it for SVGs.

**Comment by RaphaelAlvez** [comment]
> Changing the default of `svgReplaces` won't be pretty BTW. For every basic widget in existence we need to add `svgReplaces` with the old defaults or look at the `image` first and only do it for SVGs.

Yeah you are right. This also would also make the UI easier.

**Comment by robartsd** [comment]
You can always submit assets to add wtih color `#ff0000` intended for replacement. svgReplaces itself works with anything and someone else could change `#ff0000` to `currentColor` before it gets added to the official assets if desired.

**Comment by RaphaelAlvez** [comment]
> You can always submit assets to add wtih color `#ff0000` intended for replacement. svgReplaces itself works with anything and someone else could change `#ff0000` to `currentColor` before it gets added to the official assets if desired. 

Yeah I'm

**Comment by RaphaelAlvez** [comment]
> You can always submit assets to add wtih color `#ff0000` intended for replacement. svgReplaces itself works with anything and someone else could change `#ff0000` to `currentColor` before it gets added to the official assets if desired. 

Yeah I know. I feel this is the only path at the moment. I still want to keep the 

Just a bit sad because I think this setup I have now would be the ideal situation. Maybe we can consider this for the next update.

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3644/pr-test (or any other room on that server)

**Comment by 96LawDawg** [comment]
I'm not the best one to make a comment on this, but during dev-game-night play, this PR broke things.  Basic widget images would not load.

**Comment by ArnoldSmith86** [comment]
> I'm not the best one to make a comment on this, but during dev-game-night play, this PR broke things. Basic widget images would not load.

To be expected because of this todo:

>     * add file updater

File: client/js/widgets/basicwidget.js [hunk 1]
```diff
@@ -13,8 +13,8 @@ class BasicWidget extends Widget {
       activeFace: 0,
 
       image: '',
-      color: 'black',
-      svgReplaces: {},
+      color: '#ff0000',
+      svgReplaces: {"#ff0000": "color"},
```
Comment (robartsd) [inline client/js/widgets/basicwidget.js:17]: svgReplaces should be empty by default

**Review by robartsd — CHANGES_REQUESTED** [review]

**Comment by RaphaelAlvez** [comment]
This PR was a draft for a reason. I might only be wrong in not adding the file updater tag.
If the SVG and color default were to be changed there would definitely be a file update.
I'm still pondering which of both options is the best. 
I think the file format I'm proposing is better but making the update may be ugly.

**Comment by robartsd** [comment]
> This PR was a draft for a reason. I might only be wrong in not adding the file updater tag.
> If the SVG and color default were to be changed there would definitely be a file update.
> I'm still pondering which of both options is the best.
> I think the file format I'm proposing is better but making the update may be ugly.

Regardless of if you use `#ff0000` as the value that needs replacing in the SVG or use `currentColor` as existing assets do, svgReplaces should not have a replacement defined in the default set by basicwidget.js - the replacement should be set in the JSON for the widget using the asset. If you just add new assets, you don't need a file updater (even if the new assets use `#ff0000` as the color intended for replacement. A file updater would be required if you change the string being replaced in existing named assets.

**Comment by ArnoldSmith86** [comment]
@RaphaelAlvez I'm afraid we can't implement your idea of a default `svgReplaces`. It did sound cool that `color` would "just work" but the problem is that `svgReplaces` has a high cost and limitations:

There's no easy way to know if the asset URL in the `image` property is actually an SVG. The problem on the staging server was that it was downloading PNGs and applying `svgReplaces` to it. While the code could be changed to first detect if it is an SVG, I don't think it is worth it for this change. We would basically have to reimplement the browser's downloading, caching and handling of images and it would happen for every basic widget everywhere.

There are probably quite a few shortcuts that would make it easier than that but my gut tells me it's not worth it.

Maybe an alternative would be a new widget type for game pieces that only accepts SVGs? Or instead of `image` it would have `svg` and `png`?

And BTW, the default could also replace _both_ `"currentColor"` _and_ `"#ff0000"` by `color`.

**Comment by robartsd** [comment]
I do think that it would make sense for the JSON editor to make it extremely easy to add the code needed to replace "currentColor" and/or "#ff0000" with the value of the color property.

**Pull request closed without merging**